stdlib/public/runtime/ProtocolConformance.cpp

```diff
@@ -776,11 +776,23 @@ bool swift::_checkGenericRequirements(
 
     case GenericRequirementKind::Layout: {
       switch (req.getLayout()) {
-      case GenericRequirementLayoutKind::Class:
-        // Check whether the subject type is a class.
-        if (!subjectType->isAnyClass()) return true;
+      case GenericRequirementLayoutKind::Class: {
+        // existential types marked with @objc satisfy class requirement.
+        if (subjectType->isAnyExistentialType()) {
```

Comment:
Can you pull this logic out into a separate operation on Metadata, e.g., `Metadata::satisfiesClassConstraint`?

Reply:
Sure, will do before merging!